agreement/proposal.go

```diff
@@ -70,11 +69,17 @@ func (p unauthenticatedProposal) value() proposalValue {
 	return proposalValue{
 		OriginalPeriod:   p.OriginalPeriod,
 		OriginalProposer: p.OriginalProposer,
-		BlockDigest:      p.Digest(),
-		EncodingDigest:   crypto.HashObj(p),
+		BlockDigest:      p.Compressed().Digest(),
+		EncodingDigest:   crypto.HashObj(p.Compressed()),
```

Comment:
Avoid calling p.Compress() twice; it's not that cheap to iterate over all the transactions...